addons/sale_stock/views/sale_order_views.xml

```diff
@@ -68,6 +71,13 @@
                 <xpath expr="//field[@name='team_id']" position="after">
                     <field name="warehouse_id" options="{'no_create': True}" groups="stock.group_stock_multi_warehouses" optional="hide"/>
                 </xpath>
+                <field name="invoice_status" position="before">
+                    <field name="effective_date" invisible="1"/>
+                    <field name="delivery_status" optional="hide" widget="badge"
+                        decoration-success="commitment_date and (commitment_date &gt; datetime.datetime.combine(datetime.date.today(), datetime.time(23,59,59)).to_utc().strftime('%Y-%m-%d %H:%M:%S') or delivery_status=='full' and effective_date &lt;= commitment_date)"
+                        decoration-danger="commitment_date and (commitment_date &lt; datetime.datetime.combine(datetime.date.today(), datetime.time(0,0,0)).to_utc().strftime('%Y-%m-%d %H:%M:%S') or delivery_status=='full' and effective_date &gt; commitment_date)"
```

Comment:
so as long as the commitment_date is earlier than today, it will show in red color. Even when the effective_date is earlier than the commitment_date. I guess this is not correct.

Reply:
right